src/CLInterface.py

```diff
@@ -51,30 +53,23 @@ def prompt(self, options: list) -> str:
     def welcome_user(self) -> None:
         os.system("clear")
         print(welcome_message)
-
         options = list(self.modules.keys()) + ["quit"]
         request = self.prompt(options)
-
-        if request == "quit":
+        result = ""
+        if request in self.modules:
+            try:
+                result = self.modules[request].run()
+            except Exception as e:
+                self.error(f"\rerror: {e}")
+            self.show_result(result=result)
+        elif request == "quit":
```

Comment:
Looks like the merging caused the old code to be used, could you update that to the new version?

Reply:
you mean the one on main?